src/app/server/EnhancedSetupFlowProvider.h

```diff
@@ -0,0 +1,89 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <stdint.h>
+
+#include <lib/core/CHIPError.h>
+
+namespace chip {
+namespace app {
+
+/**
+ * This class provides access to the state of the Enhanced Setup Flow feature.
+ */
+class EnhancedSetupFlowProvider
+{
+public:
+    virtual ~EnhancedSetupFlowProvider() = default;
+
+    /**
+     * @param[out] outAccepted true if the required acknowledgements have been accepted, false otherwise.
+     */
+    virtual CHIP_ERROR HasTermsAndConditionsRequiredAcknowledgementsBeenAccepted(bool & outAccepted) const = 0;
```

Comment:
is the logic in this method something that a device maker might override? It seems like this is per spec, but maybe I'm misunderstand. Please add descriptive comment explaining why this logic is here rather than in general-commissioning-server.cpp, for example.

Reply:
I removed the ESF provider.